api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralQueries.java

```diff
@@ -1,6 +1,135 @@
 package org.openmrs.module.eptsreports.reporting.library.cohorts;
 
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.text.StringSubstitutor;
 import org.springframework.stereotype.Component;
 
 @Component
-public class ResumoTrimestralQueries {}
+public class ResumoTrimestralQueries {
+
+  /**
+   * Number of patients with ART suspension during the current month with LAST patient state
+   *
+   * @return String
+   */
+  public static String getPatientsWhoSuspendedTreatment(
+      boolean useBothDates,
+      int art,
+      int suspendedState,
+      int adultSeg,
+      int masterCard,
+      int artStateOfStay,
+      int preArtStateOfStay,
+      int suspendedConcept,
+      int fila,
+      int mcDrugPickup,
+      int drugPickup) {
+
+    Map<String, Integer> valuesMap = new HashMap<>();
+    valuesMap.put("art", art);
+    valuesMap.put("suspendedState", suspendedState);
+    valuesMap.put("adultSeg", adultSeg);
+    valuesMap.put("masterCard", masterCard);
+    valuesMap.put("artStateOfStay", artStateOfStay);
+    valuesMap.put("preArtStateOfStay", preArtStateOfStay);
+    valuesMap.put("suspendedConcept", suspendedConcept);
+    valuesMap.put("fila", fila);
+    valuesMap.put("mcDrugPickup", mcDrugPickup);
+    valuesMap.put("drugPickup", drugPickup);
+
+    StringBuilder sql = new StringBuilder();
+    sql.append("SELECT patient_id ");
+    sql.append("FROM (SELECT patient_id, ");
+    sql.append("             suspended_date ");
+    sql.append("      FROM (SELECT p.patient_id, ");
+    sql.append("                   Max(ps.start_date) suspended_date ");
+    sql.append("            FROM patient p ");
+    sql.append("                     JOIN patient_program pp ");
+    sql.append("                          ON p.patient_id = pp.patient_id ");
+    sql.append("                     JOIN patient_state ps ");
+    sql.append("                          ON pp.patient_program_id = ps.patient_program_id ");
+    sql.append("            WHERE p.voided = 0 ");
+    sql.append("              AND pp.voided = 0 ");
+    sql.append("              AND pp.program_id = ${art} ");
+    sql.append("              AND pp.location_id = :location ");
+    sql.append("              AND ps.voided = 0 ");
+    sql.append("              AND ps.state = ${suspendedState} ");
+    if (useBothDates) {
```

Comment:
Please remove the `if` block and `useBothDates` variable since its value is always `true`.